handbook/engineering/distribution/goals.md

```diff
@@ -2,41 +2,83 @@
 
 Performing upgrades to deployments is currently a complicated process that requires keeping a fork of our configuration and resolving diff conflicts when performing upgrades which are often complicated as the configuration might contain environment-specific customization. This process creates a bad experience for our customers because of the unknown amount of effort of the upgrade process.
 We will start by looking at our Kubernetes deployment and working on an easier update process.
 
-- **Owner**: Distribution Team
+- **Owner**: Geoffrey, Uwe
 - **Status**: In Progress
+- **Length**: ~4 months
 - **Outcomes**:
-  - Upgrades to deployments do not require resolving diff conflicts from upstream
-  - Upgrading a deployment configuration requires less than 2 hours of work
+  - Upgrades to deployments do not require resolving diff conflicts from upstream.
+  - Upgrading a deployment configuration requires less than 2 hours of work.
+- **Milestones**:
+  - ~~Research and evaluate possible tools.~~ Done: 3.19
+    - ~~Bash/JQ.~~
+    - ~~Cue.~~ Done: 3.17
+    - ~~Dhall.~~ Done: 3.18
+  - ~~Create a Kubernetes deployment PoC.~~ Done: 3.19
+  - Design Dhall Architecture. **In progress**
+  - Migrate internal deployments to Dhall, onboard other teams on how to make changes with Dhall and collect feedback. _Estimated: 3.21/3.22_
+  - Create customer facing migration path and how-to documentation. _Estimated: 3.21/3.22_
+  - Use Dhall to generate public Kubernetes manifests.
+  - Migrate a highly technical customer to Dhall and collect usage feedback.
+  - Design customization workflow.
+    - Potentially Kustomize would still be used for last-mile changes and non-standard derivations.
+
 
 ### Improve the debugging and troubleshooting process
 As we deploy Sourcegraph to multiple different environments, we need to provide a consistent and straightforward process to debug issues. We are currently lacking tools to collect debugging information (configuration, type, size, diff from upstream, etc) consistently and a process to capture the output of debugging sessions to feed back into our priorities and documentation.
 We will initially focus on reducing the time it takes to collect troubleshooting information.
 
 - **Owner**: TBD
 - **Status**: Not started
+- **Length**: TBD
 - **Outcomes**:
-  - We can categorize and capture the amount of effort spent on different incident types
-  - We can provide a straightforward set of tools to collect initial debugging and deployment information
+  - We can categorize and capture the amount of effort spent on different incident types.
+  - We can provide a straightforward set of tools to collect initial debugging and deployment information.
   - TBD
+- **Milestones**: TBD
+
+### [Split Cloud infrastructure into separate GCP projects](https://github.com/orgs/sourcegraph/projects/92)
+
+- **Owner**: Gonza
+- **Status**: On-hold
+- **Length**: ~3 months
```

Comment:
Are the milestone estimates still accurate? If so, then it seems like this goal is very much in progress, even if you aren't working on it literally this day/week.

```suggestion
- **Status**: In progress. Estimated complete by end of 2020.
```